packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -54,12 +56,15 @@ export class H5WasmApi extends ProviderApi {
     const h5wDataset = await this.getH5WasmEntity(dataset.path);
     assertH5WasmDataset(h5wDataset);
 
-    if (hasBoolType(dataset)) {
-      const value = h5wDataset.to_array();
+    // h5wasm returns integers for bool and BigInt for int64
+    // So we use to_array instead to have bool and numbers resp.
+    if (hasBoolType(dataset) || hasInt64Type(dataset)) {
```

Comment:
Two-fold fix:

1. Use `to_array` for `int64` as well to have `BigInt` converted to numbers.